src/ui/public/vis/vis_update.js

```diff
@@ -0,0 +1,34 @@
+const updateVisualizationConfig = (stateConfig, config) => {
+  if (!stateConfig || stateConfig.seriesParams) return;
+  if (!['line', 'area', 'histogram'].includes(config.type)) return;
+
+  // update value axis options
+  const isUserDefinedYAxis = config.setYExtents;
+  let mode = config.mode || 'normal';
+  if (['stacked', 'overlap'].includes(mode)) mode = 'normal';
+  config.valueAxes[0].scale = {
+    ...config.valueAxes[0].scale,
+    type: config.scale || 'linear',
+    setYExtents: config.setYExtents || false,
+    defaultYExtents: config.defaultYExtents || true,
+    min: isUserDefinedYAxis ? config.yAxis.min : undefined,
+    max: isUserDefinedYAxis ? config.yAxis.max : undefined,
+    mode: mode
+  };
+
+  // update series options
+  let interpolate = config.interpolate;
+  if (config.smoothLines) interpolate = 'cardinal';
```

Comment:
using ternary if-then-else (`const interpolate = config.smoothLines ? 'cardinal' : config.interpolate`;) would be  shorter and clearer imo. all other normalizations in this file are one-liners here too.